Maple2.Server.Game/Model/Skill/SkillRecord.cs

```diff
@@ -39,6 +40,20 @@ public class SkillRecord {
         Targets = new List<IActor>();
     }
 
+    public int GetSpCost() {
+        if (Metadata.Data.Consume.Stat.TryGetValue(BasicAttribute.Spirit, out long spiritCost)) {
+            return (int) spiritCost;
+        }
+        return 0;
+    }
+
+    public int GetStaCost() {
+        if (Metadata.Data.Consume.Stat.TryGetValue(BasicAttribute.Stamina, out long staminaCost)) {
+            return (int) staminaCost;
+        }
+        return 0;
+    }
+
```

Comment:
I could be wrong here but isn't getting the Sp or Stamina Cost only a one time occurance - through skill use from a player? I would think we could just move this and merge it into the SkillHandler UseSkill function

Reply:
Yeah should be I think.